pkg/controller/master/storageclass/storage_class_controller.go

```
@@ -0,0 +1,249 @@
+			toUpdate.Spec.Config.FilesystemOverhead.StorageClass[sc.Name] = cdiv1.Percent(v)
+		}
+	} else if toUpdate.Spec.Config.FilesystemOverhead != nil {
+		_, ok := toUpdate.Spec.Config.FilesystemOverhead.StorageClass[sc.Name]
+		if ok {
+			// Remove the storage class from FilesystemOverhead if the annotation is not present
+			delete(toUpdate.Spec.Config.FilesystemOverhead.StorageClass, sc.Name)
+		}
+	}
+
+	if !reflect.DeepEqual(toUpdate, cdi) {
+		_, err := h.cdiClient.Update(toUpdate)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{"storageclass": sc.Name}).
+				WithError(err).
+				Errorf("Failed to update CDI %s", util.CDIObjectName)
+			return fmt.Errorf("failed to update CDIConfig: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// syncStorageProfile updates the StorageProfile associated with the given StorageClass
+// It checks the annotations of the StorageClass and updates the StorageProfile accordingly.
+// The function handles the following annotations:
+// - cdi.harvesterhci.io/storageProfileCloneStrategy: updates the CloneStrategy in the StorageProfile.
+// - cdi.harvesterhci.io/storageProfileVolumeSnapshotClass: updates the SnapshotClass in the StorageProfile.
+// - cdi.harvesterhci.io/storageProfileVolumeModeAccessModes: updates the ClaimPropertySets in the StorageProfile.
+// If any of these annotations are not present in the StorageClass, the corresponding fields in the StorageProfile are set to nil (if applicable).
+func (h *storageClassHandler) syncStorageProfile(sc *storagev1.StorageClass) error {
+	profileName := sc.Name
+	profile, err := h.storageProfileCache.Get(profileName)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			logrus.WithFields(logrus.Fields{"storageclass": sc.Name}).
+				Warnf("StorageProfile %s not found, skipping update", profileName)
+			return nil
+		}
+		return err
+	}
+
+	toUpdate := profile.DeepCopy()
+
+	updateCloneStrategy(sc, toUpdate)
```

We likely don’t have a silent failure in this case.